include/alpaka/math/Complex.hpp

```diff
@@ -0,0 +1,540 @@
+    }
+
+    //! Muptiplication of a complex and a real number
+    template<typename T>
+    ALPAKA_FN_HOST_ACC Complex<T> operator*(Complex<T> const& lhs, T const& rhs)
+    {
+        return Complex<T>{lhs.real() * rhs, lhs.imag() * rhs};
+    }
+
+    //! Muptiplication of a real and a complex number
+    template<typename T>
+    ALPAKA_FN_HOST_ACC Complex<T> operator*(T const& lhs, Complex<T> const& rhs)
+    {
+        return Complex<T>{lhs * rhs.real(), lhs * rhs.imag()};
+    }
+
+    //! Division of two complex numbers
+    template<typename T>
+    ALPAKA_FN_HOST_ACC Complex<T> operator/(Complex<T> const& lhs, Complex<T> const& rhs)
+    {
+        return Complex<T>{
+            (lhs.real() * rhs.real() + lhs.imag() * rhs.imag()) / (rhs.real() * rhs.real() + rhs.imag() * rhs.imag()),
+            (lhs.imag() * rhs.real() - lhs.real() * rhs.imag()) / (rhs.real() * rhs.real() + rhs.imag() * rhs.imag())};
+    }
+
+    //! Division of complex and a real number
+    template<typename T>
+    ALPAKA_FN_HOST_ACC Complex<T> operator/(Complex<T> const& lhs, T const& rhs)
+    {
+        return Complex<T>{lhs.real() / rhs, lhs.imag() / rhs};
+    }
+
+    //! Division of a real and a complex number
+    template<typename T>
+    ALPAKA_FN_HOST_ACC Complex<T> operator/(T const& lhs, Complex<T> const& rhs)
+    {
+        return Complex<T>{
+            lhs * rhs.real() / (rhs.real() * rhs.real() + rhs.imag() * rhs.imag()),
+            -lhs * rhs.imag() / (rhs.real() * rhs.real() + rhs.imag() * rhs.imag())};
+    }
+
+    //! Equality of two complex numbers
+    template<typename T>
+    constexpr ALPAKA_FN_HOST_ACC bool operator==(Complex<T> const& lhs, Complex<T> const& rhs)
+    {
+        return math::floatEqualExactNoWarning(lhs.real(), rhs.real())
+            && math::floatEqualExactNoWarning(lhs.imag(), rhs.imag());
+    }
+
+    //! Equality of a complex and a real number
+    template<typename T>
+    constexpr ALPAKA_FN_HOST_ACC bool operator==(Complex<T> const& lhs, T const& rhs)
+    {
+        return math::floatEqualExactNoWarning(lhs.real(), rhs)
+            && math::floatEqualExactNoWarning(lhs.imag(), static_cast<T>(0));
+    }
+
+    //! Equality of a real and a complex number
+    template<typename T>
+    constexpr ALPAKA_FN_HOST_ACC bool operator==(T const& lhs, Complex<T> const& rhs)
+    {
+        return math::floatEqualExactNoWarning(lhs, rhs.real())
+            && math::floatEqualExactNoWarning(static_cast<T>(0), rhs.imag());
+    }
+
+    //! Inequality of two complex numbers.
+    //!
+    //! @note this and other versions of operator != should be removed since C++20, as so does std::complex
+    template<typename T>
+    constexpr ALPAKA_FN_HOST_ACC bool operator!=(Complex<T> const& lhs, Complex<T> const& rhs)
+    {
+        return !math::floatEqualExactNoWarning(lhs.real(), rhs.real())
+            || !math::floatEqualExactNoWarning(lhs.imag(), rhs.imag());
```

Comment:
```suggestion
        return !(lhs == rhs);
```
Reuse `==` when implementing `!=`. Applies to the operator `!=` implementations as well.